backend/routes/signatureRoutes.js

```diff
@@ -20,19 +22,43 @@ module.exports = app => {
     );
   });
 
+  app.get('/api/signatures/users/:userId', async (req, res) => {
+    const signature = await Signature.findOne({ _userID: req.params.userId });
+
+    res.send(signature);
+  });
+
   app.post('/api/signatures', async (req, res) => {
     const { user_id, campaign_id, keepUpdated } = req.body;
 
-    const signature = new Signature({
-      _userID: user_id,
-      _campaignID: campaign_id,
-      _keepUpdated: keepUpdated
+    const currentSignature = await Signature.findOne({
+      _userID: user_id
     });
 
-    try {
-      const data = await signature.save();
+    if (currentSignature) {
+      res.status(422).send({ data: 'user can only sign one campaign' });
+    } else {
+      try {
+        const signature = new Signature({
+          _userID: user_id,
+          _campaignID: campaign_id,
+          _keepUpdated: keepUpdated
+        });
+        const data = await signature.save();
 
-      res.send(data);
+        res.send(data);
+      } catch (err) {
+        res.status(422).send(err);
+      }
+    }
+  });
+
+  app.delete('/api/signatures/:sigId', async (req, res) => {
+    try {
+      await Signature.deleteOne({
+        _id: req.params.sigId
```

Comment:
I know there is some legacy code in here but might be worthwhile to clean up '_id' to reflect 'sigId' and 'userId' where applicable.